src/Shared/ProcessExtensions.cs

```diff
@@ -77,13 +77,24 @@ private static void GetAllChildIdsUnix(int parentId, ISet<int> children)
 
         private static void KillProcessUnix(int processId)
         {
-            RunProcessAndWaitForExit(
-                "kill",
-                $"-TERM {processId}",
-                out string _);
+            try
+            {
+                using Process process = Process.GetProcessById(processId);
+                process.Kill();
+            }
+            catch (ArgumentException)
+            {
+                // Process already terminated.
```

Comment:
I'm wondering if it would make sense to look for processes with parent id == processId in this case and kill those. Not sure. Also, why are there two exceptions for one error case?

Reply:
> I'm wondering if it would make sense to look for processes with parent id == processId in 
That's what `GetAllChildIdsUnix` does.

> Also, why are there two exceptions for one error case?

`ArgumentException` is thrown by `GetProcessById`.
`InvalidOperationException` is thrown by `Kill` (pre .NET 3.0).